vignettes/using-delayed-data-advanced.Rmd

```diff
@@ -408,26 +412,59 @@ if (interactive()) {
 data <- cdisc_data(
   cdisc_dataset(
     dataname = "ADSL",
-    synthetic_cdisc_dataset(dataset_name = "adsl", archive_name = "latest"),
-    code = "ADSL <- synthetic_cdisc_dataset(dataset_name = \"adsl\", archive_name = \"latest\")"
+    example_cdisc_data(dataname = "ADSL"),
+    code = "ADSL <- example_cdisc_data(dataname = \"ADSL\")"
   ),
   cdisc_dataset(
     dataname = "ADTTE",
-    synthetic_cdisc_dataset(dataset_name = "adtte", archive_name = "latest"),
-    code = "ADTTE <- synthetic_cdisc_dataset(dataset_name = \"adtte\", archive_name = \"latest\")"
+    example_cdisc_data("ADTTE"),
+    code = "ADTTE <- example_cdisc_data(dataname = \"ADTTE\")"
   )
 )
 ```
 
 One can combine multiple delayed data objects using `[teal|cdisc]_data` functions and include them in a shiny application. `TealData` gathers all objects and sets combined UI with shiny inputs from all objects. In the code chunk below we specified four dataset connectors, but it works with arbitrary many objects of any combination of the classes `TealDataset`, `TealDatasetConnector` and / or `TealDataConnector`.
 
 ```{r}
-adsl <- scda_cdisc_dataset_connector("ADSL", "adsl")
-adae <- scda_cdisc_dataset_connector("ADAE", "adae")
-advs <- scda_cdisc_dataset_connector("ADVS", "advs")
-adtte <- scda_cdisc_dataset_connector("ADTTE", "adtte")
+# nolint start
+pull_adsl <- function(ADSL) ADSL <- teal.data::rADSL
+adsl <- dataset_connector(
+  dataname = "ADSL",
+  pull_callable = callable_function(fun = pull_adsl) %>%
+    set_args(list(ADSL = as.name("ADSL"))),
+  keys = get_cdisc_keys("ADSL"),
+  label = "ADSL connector"
+)
+
+pull_adae <- function(ADAE) ADSL <- teal.data::rADAE
+adae <- dataset_connector(
+  dataname = "ADAE",
+  pull_callable = callable_function(fun = pull_adae) %>%
+    set_args(list(ADAE = as.name("ADAE"))),
+  keys = get_cdisc_keys("ADAE"),
+  label = "ADAE connector"
+)
+
+pull_advs <- function(ADVS) ADVS <- teal.data::rADVS
+advs <- dataset_connector(
+  dataname = "ADVS",
+  pull_callable = callable_function(fun = pull_advs) %>%
+    set_args(list(ADVS = as.name("ADVS"))),
+  keys = get_cdisc_keys("ADVS"),
+  label = "ADVS connector"
+)
+
+pull_adtte <- function(ADTTE) ADTTE <- teal.data::rADTTE
+adtte <- dataset_connector(
+  dataname = "ADTTE",
+  pull_callable = callable_function(fun = pull_adtte) %>%
+    set_args(list(ADTTE = as.name("ADTTE"))),
+  keys = get_cdisc_keys("ADTTE"),
+  label = "ADTTE connector"
+)
```

Comment:
There is a `cdisc_dataset_connector` which can shorten this code. `pull_adsl` also can be simplified by the `example_cdisc_data`.

```suggestion
adsl <- cdisc_dataset_connector(
  dataname = "ADSL",
  pull_callable = callable_function(function() example_cdisc_data("ADSL")),
  label = "ADSL connector"
)

adae <- cdisc_dataset_connector(
  dataname = "ADAE",
  pull_callable = callable_function(function() example_cdisc_data("ADAE")),
  label = "ADAE connector"
)

advs <- cdisc_dataset_connector(
  dataname = "ADVS",
  pull_callable = callable_function(function() example_cdisc_data("ADVS")),
  label = "ADVS connector"
)

adtte <- dataset_connector(
  dataname = "ADTTE",
  pull_callable = callable_function(function() example_cdisc_data("ADTTE")),
  label = "ADTTE connector"
)
```